File: Java/704.binary-search.java
/*
 * @lc app=leetcode id=704 lang=java
 *
 * [704] Binary Search
 */

// @lc code=start
class Solution {
    public int search(int[] nums, int target) {
        int left = 0, right = nums.length;
        while (left < right) {
            int index = (right - left) / 2 + left;
            int curr = nums[index];
            if (curr == target) {
                return index;
            } else if (curr < target) {
                left = index + 1;
            } else {
                right = index;
            }
        }
        return -1;
    }
}
// @lc code=end

File: Python/704.binary-search.py
#
# @lc app=leetcode id=704 lang=python3
#
# [704] Binary Search
#

# @lc code=start
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums)
        while l < r:
            m = (l + r) // 2
            if nums[m] < target:
                l = m + 1
            elif nums[m] > target:
                r = m
            else:
                return m
        return -1

# @lc code=end

